### DISCUSSION TITLE
Waveguided FEL

### DISCUSSION THREAD

**Discussion by sonafel** [svenreiche/Genesis-1.3-Version4#87] [Q&A] [answered]
For THz FEL, wave guiding in the resonator regime will be a good choice to avoid the diffraction losses at longer wavelengths. 
Does Genesis 1.3 version 4 support wave guiding in the resonator regime?

**Comment by svenreiche** [comment, marked as answer]
Unfortunately Genesis 1.3 Version 4 does not support waveguide modes. the core algorithm is finite differences for the field equation, while waveguide modes are finite mode solver. It will be a very time-consuming task to support this. I know from research groups, (e.g. Prof Pietro Musumeci, UCLA) which have an inofficial verison of Genesis Version 2 to support waveguides.  Also they are using the FEL plugin for GPT to solve this problems.  I hope this helps.

Best,
Sven
